lib/workers/repository/update/pr/code-owners.ts

```diff
@@ -21,37 +91,71 @@ export async function codeOwnersForPr(pr: Pr): Promise<string[]> {
 
     logger.debug(`Found CODEOWNERS file: ${codeOwnersFile}`);
 
+    // Get list of modified files in PR
     const prFiles = await getBranchFiles(pr.sourceBranch);
-    const rules = codeOwnersFile
+
+    if (!prFiles?.length) {
+      logger.debug('PR includes no files');
+      return [];
+    }
+
+    // Convert CODEOWNERS file into list of matching rules
+    const fileOwnerRules = codeOwnersFile
       .split(newlineRegex)
+      // Remove empty and commented lines
       .map((line) => line.trim())
       .filter((line) => line && !line.startsWith('#'))
-      .map((line) => {
-        const [pattern, ...usernames] = line.split(regEx(/\s+/));
-        return {
-          usernames,
-          match: (path: string) => {
-            const matcher = ignore().add(pattern);
-            return matcher.ignores(path);
-          },
-        };
-      })
-      .reverse();
+      // Extract pattern & usernames
+      .map(extractOwnersFromLine);
+
     logger.debug(
-      { prFiles, rules },
+      { prFiles, fileOwnerRules },
       'PR files and rules to match for CODEOWNERS'
     );
-    const matchingRule = rules.find((rule) => prFiles?.every(rule.match));
-    if (!matchingRule) {
-      logger.debug('No matching CODEOWNERS rule found');
-      return [];
-    }
+
+    // Apply rules & get list of owners for each prFile
+    const emptyRules = fileOwnerRules.filter(
+      (rule) => rule.usernames.length === 0
+    );
+    const fileOwners =
+      // Map through all prFiles and match said file(s) with all the rules
+      prFiles
+        .map((file) => matchFileToOwners(file, fileOwnerRules))
+
+        // Match file again but this time only with emptyRules, to ensure that files which have no owner set remain owner-less
+        .map((fileMatch) => {
+          const matchEmpty = emptyRules.find((rule) =>
+            rule.match(fileMatch.file)
+          );
+          if (matchEmpty) {
+            return { ...fileMatch, userScoreMap: new Map<string, number>() };
+          }
+          return fileMatch;
+        });
+
+    logger.debug(
+      `CODEOWNERS matched the following files: ${JSON.stringify(fileOwners)}`
```

Comment:
did you test how maps are serialized?

Reply:
Just pushed e222e7e76c8f4940f62a979bd658bf00509f16ec with a fix, turns out Maps and stringify don't play along.